### DISCUSSION TITLE
Is it possible to modify the background/quit notifications before displaying them on the phone?

### DISCUSSION THREAD

**Discussion by HosseinRashno** [invertase/react-native-firebase#7499] [Q&A] [answered]
First of all, I want to thank the maintainers of this package for their wonderful work on this package.   
  
I use `"@react-native-firebase/messaging": "^18.3.2"` in my app to handle push notifications and everything works as expected.  Now I have a feature request from the client and I can't talk them out of it because they already have this feature in one of their other apps.  
  
They want to send personalized notifications. So basically they want to send a notification like this:
```json
{
  "title": "Hello @name!",
  "body": "Hey @name, you can check out our new stuff on the app"
}
```
And then from the app side, I have to replace `@name` with the user's real name. (`@name` is just an example.).  
  
I know I can achieve this for foreground notifications by using Notifee. Still, I can't figure out how to do this for background/quit notifications that are handled automatically by `@react-native-firebase/messaging`.

Could you give me a clue as to how to implement this feature?   
  
Also, I was thinking maybe I should make the server to only send data notifications and handle everything by the app. But then `setBackgroundMessageHandler` is unreliable on iOS and there is a good chance that the users miss the notification.

I use the following packages in my app:
```json
{
    "react": "18.2.0",
    "react-native": "0.72.1",
    "@react-native-firebase/app": "^18.3.2",
    "@react-native-firebase/messaging": "^18.3.2"
}
```

**Comment by mikehardy** [comment, marked as answer]
Notifee has a Notification Extension Helper feature that I believe you want to use in combination with a `notification` block on iOS since data-only messages are unreliable there

On Android you will likely have okay delivery performance with data-only notifications so can simply use data-only notifications then style+display notifications as normal via Notifee APIs

**Comment by HosseinRashno** [comment]
Thanks a lot for the app. The suggested solution worked for me.
